### PR TITLE
Test Explorer - Part 1

### PR DESCRIPTION
**Pull request by xisui-MSFT**
Added a Test Explorer for CTest using the [Test API](https://code.visualstudio.com/api/extension-guides/testing). For detailed design decisions, please review the document shared internally.
#1682

### CODE REVIEW

File: src/cmakeTaskProvider.ts [hunk 1]
```diff
@@ -518,13 +518,13 @@ export class CustomBuildTaskTerminal implements vscode.Pseudoterminal, proc.Outp
...
                 }
             }
-            const result: number | null | undefined = cmakeDriver ? await project?.runCTestCustomized(cmakeDriver, testPreset, this) : undefined;
-            if (result === undefined || result === null) {
+            const result: number | undefined = cmakeDriver ? await project?.runCTestCustomized(cmakeDriver, testPreset, this) : undefined;
+            if (result === undefined) {
                 this.writeEmitter.fire(localize('ctest.run.terminated', 'CTest run was terminated') + endOfLine);
                 this.closeEmitter.fire(-1);
             } else {
-                this.writeEmitter.fire(localize('ctest.finished.with.code', 'CTest finished with return code {0}', result) + endOfLine);
```
Comment (benmcmorran) [inline src/cmakeTaskProvider.ts:526]: Would it be useful to continue to log the exit code for diagnosing CTest issues?

**Review by benmcmorran — APPROVED** [review]

**Comment by benmcmorran** [comment]
I'm not sure if you were planning to have one or multiple CHANGELOG entries, but if you're planning to have multiple, please edit CHANGELOG.md to describe the new commands added in this PR.

**Comment by xisui-MSFT** [comment]
> I'm not sure if you were planning to have one or multiple CHANGELOG entries, but if you're planning to have multiple, please edit CHANGELOG.md to describe the new commands added in this PR.

I'll change CHANGELOG in part 3 for convenience.

File: src/cmakeTaskProvider.ts (hunk 1 above)
Comment (xisui-MSFT) [reply src/cmakeTaskProvider.ts:526]: The original error is the exit code of ctest.exe. But now we invoke ctest for each test run, this exit code doesn't make sense anymore, and so this number returned just shows succeeded/failed status. But this status should be very obvious in the test explorer. And therefore, the exit code is removed here.

**Pull request merged**